app/scripts/controllers/permissions/restrictedMethods.js

```diff
@@ -1,12 +1,28 @@
-export default function getRestrictedMethods (permissionsController) {
+export default function getRestrictedMethods ({ getIdentities, getKeyringAccounts }) {
   return {
 
     'eth_accounts': {
       description: `View the addresses of the user's chosen accounts.`,
       method: (_, res, __, end) => {
-        permissionsController.getKeyringAccounts()
+        getKeyringAccounts()
           .then((accounts) => {
+            const identities = getIdentities()
             res.result = accounts
+              .sort((firstAddress, secondAddress) => {
+                if (!identities[firstAddress]) {
+                  throw new Error(`Missing identity for address ${firstAddress}`)
+                } else if (!identities[secondAddress]) {
+                  throw new Error(`Missing identity for address ${secondAddress}`)
+                } else if (identities[firstAddress].lastSelected === identities[secondAddress].lastSelected) {
+                  return 0
+                } else if (identities[firstAddress].lastSelected === undefined) {
+                  return 1
+                } else if (identities[secondAddress].lastSelected === undefined) {
+                  return -1
+                }
+
+                return identities[secondAddress].lastSelected - identities[firstAddress].lastSelected
+              })
```

Comment:
Since index 0 of the accounts array is the only one with any meaning, do we need to sort the whole thing?

The alternative would be to find the `Math.min` of the identities by their `lastSelected` property, put the corresponding account first, if any, and concatenate the rest.

Reply:
>Ah, no we can't, because the addresses are filtered by the caveats after this call.

Yep, exactly. So because of how the caveat filter works, we're kinda forced to order them by last-selected time.

Reply:
Also, several dozen things will break before the sorting becomes a performance problem. This is definitely fine for now. It'd be nice if this sorting didn't have to happen every call to `eth_accounts`, but we can figure that out later.